membership-common/src/main/scala/com/gu/memsub/subsv2/Plan.scala

```diff
@@ -122,178 +108,121 @@ case object Years extends UpToPeriodsType {
+          case ZSemiAnnual => Validation.success[String, BillingPeriod](SixMonthsRecurring)
+          case _ => Validation.f[BillingPeriod](s"zuora recurring period was $zPeriod")
+        }
+      case (OneTime, None) => Validation.success[String, BillingPeriod](OneTimeChargeBillingPeriod) // This represents a one time rate plan charge
+      case _ =>
+        Validation.f[BillingPeriod](
+          s"period =${zBillingPeriod} specificBillingPeriod=${specificBillingPeriod} uptoPeriodsType=${upToPeriodsType}, uptoPeriods=${upToPeriods}",
+        )
+    }
+
 }
 
 /** Low level model of a rate plan, as it appears on a subscription in Zuora
   */
-case class SubscriptionZuoraPlan(
+case class RatePlan(
     id: RatePlanId,
     productRatePlanId: ProductRatePlanId,
     productName: String,
     lastChangeType: Option[String],
     features: List[Feature],
     chargedThroughDate: Option[LocalDate],
-    charges: NonEmptyList[ZuoraCharge],
+    ratePlanCharges: NonEmptyList[RatePlanCharge],
     start: LocalDate,
     end: LocalDate,
 ) {
-  def price = charges.list.toList.flatMap { charge =>
-    charge.pricing.prices.map(_.amount)
-  }.sum
-}
 
-/** Low level model of a product rate plan, as it appears in the Zuora product catalog
-  */
-case class CatalogZuoraPlan(
-    id: ProductRatePlanId,
-    name: String,
-    description: String,
-    productId: ProductId,
-    saving: Option[String],
-    charges: List[ZuoraCharge],
-    benefits: Map[ProductRatePlanChargeId, Benefit],
-    status: Status,
-    frontendId: Option[FrontendId],
-    private val productTypeOption: Option[String],
-) {
-  lazy val productType = productTypeOption.getOrElse(throw new RuntimeException("Product type is undefined for plan: " + name))
-}
-
-sealed trait FrontendId {
-  def name: String
-}
-object FrontendId {
-
-  case object OneYear extends FrontendId { val name = "OneYear" }
-  case object ThreeMonths extends FrontendId { val name = "ThreeMonths" }
-  case object Monthly extends FrontendId { val name = "Monthly" }
-  case object Quarterly extends FrontendId { val name = "Quarterly" }
-  case object Yearly extends FrontendId { val name = "Yearly" }
-  case object Introductory extends FrontendId { val name = "Introductory" }
-  case object SixWeeks extends FrontendId { val name = "SixWeeks" }
-
-  val all = List(OneYear, ThreeMonths, Monthly, Quarterly, Yearly, Introductory, SixWeeks)
-
-  def get(jsonString: String): Option[FrontendId] =
-    all.find(_.name == jsonString)
+  def totalChargesMinorUnit: Int =
+    ratePlanCharges.map(c => (c.pricing.prices.head.amount * 100).toInt).list.toList.sum
+
+  def chargesPrice: PricingSummary =
+    ratePlanCharges
+      .map(_.pricing)
+      .list
+      .toList
+      .reduce((f1, f2) =>
+        PricingSummary(
+          f1.underlying.keySet.intersect(f2.underlying.keySet).map(c => c -> Price(f1.underlying(c).amount + f2.underlying(c).amount, c)).toMap,
```

Comment:
this logic is from PricingSemigroup which is now deleted above - membership-common/src/main/scala/com/gu/memsub/PricingSummary.scala